src/coreclr/tools/aot/ILCompiler.Diagnostics/PerfMapWriter.cs

```diff
@@ -72,6 +62,32 @@ public static void Write(string perfMapFileName, int perfMapFormatVersion, IEnum
             }
         }
 
+        private static void WritePerfMapV1Header(TargetDetails details, PerfMapWriter perfMapWriter, Guid outputGuid)
+        {
+            // Make sure these get emitted in this order, other tools in the ecosystem like the symbol uploader and PerfView rely on this.
+            // In particular, the order of it. Append only.
+            perfMapWriter.WriteLine(outputGuid.ToString(), (uint)PseudoRVA.OutputGuid, 0);
+            perfMapWriter.WriteLine(CurrentFormatVersion.ToString(), (uint)PseudoRVA.FormatVersion, 0);
+            perfMapWriter.WriteLine(details.OperatingSystem.ToString(), (uint)PseudoRVA.TargetOS, 0);
```

Comment:
Do we have any safeguards that if someone decided to rename the enum values (for example because they're tired of seeing OSX because Apple decided to rebrand the OS back to MacOS), diagnostics doesn't become broken?

However much I hate duplication, these enums are internal to the type system and if we need to use them in a file format, they should be mapped to stable identifiers first.

Reply:
You are right - I was intending to emit the numeric values now. However, even that seems unstable (reordering them or deleting values would mess things up).

Reply:
The way we deal with that, is we define enums as part of the ReadyToRunConstants.cs file ... and then those are the enums we use. In general we don't change any of those values, we just add. (Although, we have a version field so we could change them if we really wanted to.) Please don't write code that assumes that the general TypeSystem constants are constant though. That's not OK, as we reserve the right to change them.